helix-view/src/clipboard.rs

```diff
@@ -81,7 +81,27 @@ pub fn get_clipboard_provider() -> Box<dyn ClipboardProvider> {
         return Box::new(provider::WindowsProvider);
 
         #[cfg(not(target_os = "windows"))]
-        return Box::new(provider::NopProvider);
+        return Box::new(provider::NopProvider { buf: String::new() });
+    }
+}
+
+pub fn get_primary_selection_provider() -> Box<dyn ClipboardProvider> {
+    // TODO: support for user-defined provider, probably when we have plugin support by setting a
+    // variable?
+
+    if env_var_is_set("DISPLAY") && exists("xsel") && is_exit_success("xsel", &["-o"]) {
```

Comment:
Here I preferred X11 intentionally. Reason: `wl-copy`/`wl-paste` create surfaces which make terminal unfocused for some milliseconds, which is annoying since terminal blinks. Native Wayland would be preferred when Helix will be a windowed Wayland application, so it would be better for clipboard operations.